### ISSUE TITLE
[ANDROID] Keyboard navigation and focus does not work properly for SectionList

### ISSUE THREAD

**Issue by ewc003** [ArturKalach/react-native-a11y#97]
Hello,

I am having an issue using the tab key to navigate through elements in the SectionList [Android OS].
My renderItem function returns a Pressable component similar to the example from [here](https://github.com/ArturKalach/react-native-a11y/blob/master/examples/A11ySample/src/components/Button/Button.tsx). Focus state seems to be updating correctly because style changes but when selecting an item from the list, a different item is selected.

There are 2 cases I noticed.
1. Using the tab key does not update the selected item. The first item in my list is always selected, regardless of how many times I press 'tab'.
2. Using directional key "down" updates the selected item but is offset by x items (the item focused when pressing down for the first time is x indices away from the first)

Thanks in advance

**Comment by ArturKalach** [comment]
Hello @ewc003,

Thank you so much for using the package and creating the issue. I really appreciate it.

Could I ask you to try another package, [react-native-external-keyboard](https://www.npmjs.com/package/react-native-external-keyboard)?

Unfortunately, the current version of react-native-a11y may have many issues with newArch—possibly not only that but other problems as well. So, the simplest solution could be to use react-native-external-keyboard.

I will try to test the issue with both packages, but it may take some time. Additionaly, it could happen that all fixes and updates will be made to `react-native-external-keyboard` package first.

--------

I will try to reproduce the issue, but it would really help if you could share an example.

-------


Regards,
Artur Kalach

**Comment by ewc003** [comment]
Hi @ArturKalach,
Here is example code for my SectionList Integration. Thanks!

```
const List = ({ onPress, accessibilityLabel, accessibilityHint, renderSectionHeader, ItemSeparatorComponent, sections, keyExtractor }) => {
    const renderItem = ({ item }) => {
        return (
            <AccessibleButton
                onPress={() => onPress(item)}
                title={item}
            />
        );
    }

    return (
        <A11yProvider>
            <KeyboardProvider value={true}>
                <View style={ this.styles.LocationListView_rootContainer }>
                    <SectionList
                        accessibilityLabel={accessibilityLabel}
                        accessibilityHint={accessibilityHint}
                        renderItem={renderItem}
                        renderSectionHeader={renderSectionHeader}
                        ItemSeparatorComponent={ItemSeparatorComponent}
                        sections={sections}
                        keyExtractor={keyExtractor}
                        stickySectionHeadersEnabled={true}
                        nestedScrollEnabled={true}
                    />
                </View>
            </KeyboardProvider>
        </A11yProvider>
    );
}

export default List;
```

**Comment by ArturKalach** [comment]
Hello @ewc003,

I checked this yesterday and was able to reproduce the issue. I will try to fix it and update the package with all the features present in other repositories, but it may take some time for refactoring, updating the documentation, and testing.

-------
So far, I would recommend trying [react-native-external-keyboard](https://www.npmjs.com/package/react-native-external-keyboard). I have already tested it, and the focus worked predictably for me.

```jsx
import React from "react";
import { SectionList, Text, View } from "react-native";
import { Pressable } from "react-native-external-keyboard";

const DATA = [
    {
      title: 'Main dishes',
      data: ['Pizza', 'Burger', 'Risotto'],
    },
    {
      title: 'Sides',
      data: ['French Fries', 'Onion Rings', 'Fried Shrimps'],
    },
    {
      title: 'Drinks',
      data: ['Water', 'Coke', 'Beer'],
    },
    {
      title: 'Desserts',
      data: ['Cheese Cake', 'Ice Cream'],
    },
  ];

  
const List = ({ onPress, accessibilityLabel, accessibilityHint, renderSectionHeader, ItemSeparatorComponent, sections, keyExtractor }) => {
    const renderItem = ({ item }) => {
        return (
            <Pressable onPress={() => onPress(item)}>
                <Text>{item}</Text>
            </Pressable>
        );
    }

    return (
                <View style={{ flex: 1}}>
                    <SectionList
                        accessibilityLabel={accessibilityLabel}
                        accessibilityHint={accessibilityHint}
                        renderItem={renderItem}
                        renderSectionHeader={renderSectionHeader}
                        ItemSeparatorComponent={ItemSeparatorComponent}
                        sections={sections}
                        keyExtractor={keyExtractor}
                        stickySectionHeadersEnabled={true}
                        nestedScrollEnabled={true}
                    />
                </View>
    );
};

export const Example = () =>  <View>
    <List
        onPress={(i) => console.log(i)}
        accessibilityLabel="Mi"
        accessibilityHint="Mimi"
        // renderSectionHeader
        sections={DATA}
        // keyExtractor
    />
</View>
```

--------

Sorry for the inconveniences, 
Artur Kalach

**Comment by ewc003** [comment]
Hi Artur,
I noticed that onKeyDownPress works as expected so will be using that instead. The issue seems to only occur when the size of the SectionList exceeds the area of the parent component.
